### ISSUE TITLE
Cannot ignore files using .git-ftp-ignore

### ISSUE THREAD

**Issue by CLL80** [git-ftp/git-ftp#166]
I have tried many times to ignore files for upload, but nothing I do is causing git-ftp to actually ignore the files I specify.

I've tried using an ignore file with a name of both ".git-ftp-ignore" and ".gitftpignore". I've added that file as the ignore file for my specific branch:
````
    [branch "dev"]
    gitftpignore=.git-ftp-ignore
````

Nothing I do causes the files and patterns I specify in this file to actually be ignored for upload.

**Comment by mkllnk** [comment]
Is the file .git-ftp-ignore in your repository's root directory (where .git/ is)?

How does your .git-ftp-ignore file look like (content)?

Which version of git-ftp are you using?

**Comment by CLL80** [comment]
Thanks for the response.

Yes, it sits alongside the .git directory at the same level as .gitignore.

Here is the content of that file:
````
    #/html/php.ini

    #/html/.htaccess.dev
    /html/blog/.htaccess
    /html/_tpl/.htaccess
    /html/_inc/.htaccess

    *.dev
````

It doesn't ignore any of those files, even if I don't use a wildcard and uncomment ` #/html/.htaccess.dev`.

I'm using version 0.9.0 on Mavericks.

**Comment by mkllnk** [comment]
Unfortunately, .git-ftp-ignore doesn't use the same syntax as .gitignore. The version you use interprets each line as regular expression applied to a list of filenames preceded with a status flag. In this case your file needs to look like this:

    ^. html/blog/\.htaccess$
    ^. html/_tpl/\.htaccess$
    ^. html/_inc/\.htaccess$

    \.dev$

But we changed that in the most recent version of git-ftp. There you would use shell glob patterns.

**Comment by CLL80** [comment]
`\.dev$` does the trick—no .htaccess.dev files are being uploaded. I can't seem to get the .htaccess file in the html directory to be ignored based on your example:

I would like to ignore the .htaccess file here:
html/.htaccess

I've tried:

`^. html/\.htaccess$`

`^. html\/\.htaccess$`

`^html\/\.htaccess$` (which based on my knowledge of regex would be correct)

and

`^\/html\/\.htaccess$`

and more with no success.

**Comment by mkllnk** [comment]
The regex is matched against an old file list with a file status like this:
```
M dir/file1.txt
D dir/old_file.txt
```

But the list can also contain a tab instead of a space:
```
H	dir/file1.txt
H	dir/file2.txt
```

So you can try:
```
^..html/\.htaccess$
```

**Comment by CLL80** [comment]
Yes, that worked. Thanks for your help with this!

Is there any documentation that you can recommend I take a look at so I can familiarize myself with the shell glob patterns? I'd like to make sure I understand how they work so I can ignore the same files before updating git-ftp to the current version.

**Comment by mkllnk** [comment]
That's alright. Hence we changed the code to glob patterns which are very similar to .gitignore. In fact, .gitignore uses glob patterns, but adds a little bit more.

Glob patterns are just filenames with wildcards. Allowed wildcards are `*`, `?` and `[characters]`. http://man7.org/linux/man-pages/man7/glob.7.html

.gitignore has a bit more logic on directory names and patterns starting with `/`. That's all.

**Comment by WoodrowShigeru** [comment]
Could you please provide a documenation for the .git-ftp-ignore syntax? Instructions are unclear.

**Comment by LukasFritzeDev** [comment]
@WoodrowShigeru You can find a documentation with some examples in the [manual](https://github.com/git-ftp/git-ftp/blob/master/man/git-ftp.1.md#ignoring-files-to-be-synced). If this does not help, you can tell us which files you want to ignore and we can try to find a solution. “Instructions are unclear.” is a litte vague 😉

**Comment by WoodrowShigeru** [comment]
I got it to work by comparing binomialstew's input and mkllnk's answer, and applying that to my case. I don't understand *at all* why that works because that expression isn't explained anywhere, not even in the manual. But it works and I've moved on by now.

It was a surprise to learn that the git-ftp-ignore syntax *diverts* from the gitignore syntax. That bit of info is hidden somewhere in the issues instead of being pointed out in the man page. That page states how it is interpreted – but if you don't know gitignore by heart, then you don't realize that there is a difference.

To be fair, the instructions were unclear mostly because I haven't seen that manual page at the time. But even now, I still can't explain this expression. Why do I have to convert the root slash to "caret period space"?
